core/renderers/zelos/measurables/rows.js

```diff
@@ -62,7 +62,10 @@ Blockly.zelos.TopRow.prototype.endsWithElemSpacer = function() {
  * @override
  */
 Blockly.zelos.TopRow.prototype.hasLeftSquareCorner = function(block) {
-  return !!block.outputConnection;
+  var hasHat = (typeof block.hat !== 'undefined' ?
```

Comment:
Just curious why you are using typeof instead of a.hat != undefined?

Reply:
Mostly for consistency with https://github.com/google/blockly/blob/develop/core/renderers/measurables/rows.js#L293
but in general this type of check is no longer necessary (it used to be back in the day, when browsers let you change the value of ``undefined``)
Another subtle difference between the two is the typeof check makes sure it had never been defined before, rather than block.hat being set to undefined.

I'll defer the question over to @moniika as she was the one that changed it from ``block.hat`` to ``typeof block.hat !== 'undefined'`` in https://github.com/google/blockly/commit/26db9167db6da15c5ba463e345727243054bea64